tests/test_parser.py

```diff
@@ -84,44 +85,65 @@ def test_constructor_cache(self, tmpdir):
         assert df2.cached_content
 
     def test_dockerfile_structure(self, dfparser):
-        dfparser.lines = ["# comment\n",        # should be ignored
-                          " From  \\\n",        # mixed-case
-                          "   base\n",          # extra ws, continuation line
-                          " # comment\n",
-                          " label  foo  \\\n",  # extra ws
-                          "    # comment\n",    # should be ignored
-                          "    bar  \n",        # extra ws, continuation line
+        dfparser.lines = ["# comment\n",                # single-line comment
+                          " From  \\\n",                # mixed-case
+                          "   base\n",                  # extra ws, continuation line
+                          " #    another   comment\n",  # extra ws
+                          " label  foo  \\\n",          # extra ws
+                          "# multi-line \n",            # multi-line comment, interrupt instruction
+                          "# ...comment\n",
+                          "    bar  \n",                # extra ws, instruction continuation
                           "USER  {0}\n".format(NON_ASCII),
-                          "# comment \\\n",     # extra ws
-                          "# comment \\ \n",    # extra ws with a space
-                          "# comment \\\\ \n",  # two backslashes
+                          "# comment \\\n",             # extra ws
+                          "# with \\ \n",               # extra ws with a space
```

Comment:
I miss test cases for:
* `#comment` - comment without spaces
* `# comment with # inside`

Reply:
:+1: